src/App.tsx

```diff
@@ -1,61 +1,147 @@
+
+  const [shouldErrorOnUserSelect, setErrorOnUserSelect] = useState(false);
+  const [shouldErrorOnTitleInput, setErrorOnTitleInput] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    if (shouldErrorOnTitleInput === true) {
+      setErrorOnTitleInput(false);
+    }
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.currentTarget.value);
+    if (shouldErrorOnUserSelect === true) {
+      setErrorOnUserSelect(false);
+    }
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    setErrorOnTitleInput(!title.trim());
+    setErrorOnUserSelect(!selectedUserId);
+
+    if (title.trim() === '' || selectedUserId === 0) {
+      setErrorOnTitleInput(title.trim() === '');
+      setErrorOnUserSelect(selectedUserId === 0);
+
+      return;
+    }
+
+    // const userToAdd = getUserById(selectedUser);
+
+    setTodos(current => {
+      const maxTodoId = Math.max(...current.map(todo => todo.id));
+
+      return [
+        ...current,
+        {
+          id: maxTodoId + 1,
+          title,
+          completed: false,
+          userId: selectedUserId,
+          user: getUserById(selectedUserId),
+        },
+      ];
+    });
+
+    setTitle('');
+    setSelectedUserId(0);
+  };
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <h1 className="title is-1">
+        Add todo form
+      </h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title"> Write down your task:</label>
+
+          <input
+            type="text"
+            className="input is-info"
+            data-cy="titleInput"
+            placeholder="Enter title of task"
+            value={title}
+            onChange={handleTitleChange}
+          />
+          {shouldErrorOnTitleInput && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label
+            htmlFor="title"
+          >
+            Choose a user:
+          </label>
+
+          <div className="select is-info">
+            <select
+              data-cy="userSelect"
+              id="userSelect"
+              name="userSelect"
+              value={selectedUserId}
+              onChange={handleUserChange}
+            >
+              <option disabled>Choose a user</option>
+
+              {usersFromServer.map(user => (
+                <option key={user.id} value={user.id}>{user.name}</option>
+              ))}
```

Comment:
![image](https://user-images.githubusercontent.com/102413088/214254889-451453a6-062c-41cd-aee0-01f4bb73f003.png)

I cannot choose the first user, can you?